web/modules/api/src/api.ts

```diff
@@ -152,6 +152,25 @@ class WikijumpAPIInstance extends Api<void> {
     return this.request<T, void>({ method: "POST", baseUrl, path, body, type })
   }
 
+  /**
+   * Executes an API request and catches any errors by returning `null`.
+   * This can be used when the result of a request is easily discarded.
+   *
+   * @param method - The API method to call.
+   * @param args - The arguments to pass to the method.
+   */
+  async try<
+    M extends keyof FilterFor<Api<void>, AnyFunction>,
+    F extends AnyFunction = FilterFor<Api<void>, AnyFunction>[M]
+  >(method: M, ...args: Parameters<F>): Promise<null | Awaited<ReturnType<F>>> {
+    try {
+      // @ts-ignore - TS can't quite figure this one out
+      return await this[method](...args)
```

Comment:
I keep hearing that "return await" is bad but I still don't understand why.

Reply:
It's kind of bad. It also kind of doesn't matter, usually.

Promises can be nested, e.g. `Promise<Promise<Promise<string>>>`. Asynchronous functions implicitly wrap their returned value in a `Promise`. If I didn't do `await`, I would be returning `Promise<Promise<foo>>` instead of a `Promise<foo>`. I'd rather the return type be the same as the function you are "trying", so I `await` the function.

Arguably, the consumer should be the one bothering to `await` any promises, regardless of nesting. However, I don't know when this would actually matter, because a promise in JS runs its executor function regardless of if you're waiting on it or not. `await` blocks if the executor isn't resolved yet, or immediately resolves if it is.